tests/graph/test-data/rust/src/seeding/producer/data_type.rs

```diff
@@ -46,76 +53,488 @@ pub enum DataTypeProducerConfigError {
+            items.push((
+                Arc::clone(domain),
+                Arc::<str>::from(user.shortname.as_str()),
+                WebId::from(user.id),
+            ));
+        }
+        Self { items }
+    }
+}
+
+impl WebCatalog for InMemoryWebCatalog {
+    fn len(&self) -> usize {
+        self.items.len()
+    }
+
+    fn get_entry(&self, index: usize) -> Option<(Arc<str>, Arc<str>, WebId)> {
+        self.items.get(index).cloned()
+    }
+}
+
+#[derive(Debug)]
+enum BoundDomainSampler<'a, U: WebCatalog, O: WebCatalog> {
+    Remote {
+        domain: DomainDistribution,
+        fetched_at: OffsetDateTime,
+    },
+    Local(LocalChoiceSampler<'a, U, O>),
+    Mixed {
+        chooser: WeightedIndex<u32>,
+        remote: DomainDistribution,
+        fetched_at: OffsetDateTime,
+        local: LocalChoiceSampler<'a, U, O>,
+    },
+}
+
+#[derive(Debug)]
+enum SampledDomain {
+    Remote {
+        domain: Arc<str>,
+        shortname: Arc<str>,
+        fetched_at: OffsetDateTime,
+    },
+    Local {
+        domain: Arc<str>,
+        shortname: Arc<str>,
+        web_id: WebId,
+    },
+}
+
+impl<U: WebCatalog, O: WebCatalog> BoundDomainSampler<'_, U, O> {
+    fn sample<R: rand::Rng + ?Sized>(&self, rng: &mut R) -> SampledDomain {
+        match self {
+            BoundDomainSampler::Remote { domain, fetched_at } => {
+                let (domain, shortname) = domain.sample(rng);
+                SampledDomain::Remote {
+                    domain,
+                    shortname,
+                    fetched_at: *fetched_at,
+                }
+            }
+            BoundDomainSampler::Local(local) => {
+                let (domain, shortname, web_id) = local.sample(rng);
+                SampledDomain::Local {
+                    domain,
+                    shortname,
+                    web_id,
+                }
+            }
+            BoundDomainSampler::Mixed {
+                chooser,
+                remote,
+                fetched_at,
+                local,
+            } => {
+                if chooser.sample(rng) == 0 {
+                    let (domain, shortname) = remote.sample(rng);
+                    SampledDomain::Remote {
+                        domain,
+                        shortname,
+                        fetched_at: *fetched_at,
+                    }
+                } else {
+                    let (domain, shortname, web_id) = local.sample(rng);
+                    SampledDomain::Local {
+                        domain,
+                        shortname,
+                        web_id,
+                    }
+                }
+            }
+        }
+    }
+}
+
+#[derive(Clone)]
+struct LocalUniform<'a, C: WebCatalog> {
+    catalog: &'a C,
+}
+
+impl<C: WebCatalog> core::fmt::Debug for LocalUniform<'_, C> {
+    fn fmt(&self, fmt: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        fmt.debug_struct("LocalUniform").finish()
+    }
+}
```

Comment:
```suggestion
#[derive(Clone, derive_more::Debug)]
struct LocalUniform<'a, C: WebCatalog> {
	#[debug(skip)]
    catalog: &'a C,
}
```

This should also work™